src/error.rs

```diff
@@ -729,6 +730,7 @@ where
         unsafe { &mut *(self.vtable.object_mut)(self) }
     }
 
+    #[cfg(feature = "std")]
```

Comment:
I don't think we can make this change because this would mean that the `Item` type of `Chain` changes when we use `no_std` mode. This would cause disabling / enabling features to become a breaking change which is a no-go. Did you need the chain interface on no_std?

Reply:
I added the `cfg` here because the only usage of the `ErrorImpl::chain` function is here: https://github.com/yaahc/eyre/blob/a197feb4e3368f70735a2fe34990fbebb10487cd/src/error.rs#L305-L307 and is marked as `std` only. In addition `ErrorImpl::chain` is `pub crate` making it dead code. It compiles, but clippy rightfully complaints about dead code without this change:

```sh

❯ cargo check --no-default-features                                                                                                                                                                         ⇡!? master 
   Compiling eyre v0.4.2 (/Users/Matthias/git/rust/eyre)
warning: associated function is never used: `chain`
   --> src/error.rs:732:19
    |
732 |     pub(crate) fn chain(&self) -> Chain<'_> {
    |                   ^^^^^
    |
```